src/main/java/spoon/support/compiler/jdt/PositionBuilder.java

```diff
@@ -156,18 +162,45 @@ SourcePosition buildPositionCtElement(CtElement e, ASTNode node) {
 			int modifiersSourceStart = variableDeclaration.modifiersSourceStart;
 			int declarationSourceStart = variableDeclaration.declarationSourceStart;
 			int declarationSourceEnd = variableDeclaration.declarationSourceEnd;
+			if (declarationSourceStart == 0 && declarationSourceEnd == 0) {
+				return SourcePosition.NOPOSITION;
+			}
+			if (e instanceof CtCatchVariable) {
+				/* compiler delivers wrong declarationSourceStart in case like: */
+				//... catch/*2*/ ( /*3*/ final @Deprecated /*4*/ ClassCastException /*5*/ e /*6*/) /*7*/ {
+				/*
+				 * the declarationSourceStart should be after the '(', but sometime it is before
+				 * So we have to compute correct offset here
+				 */
+				CtTry tryStatement = this.jdtTreeBuilder.getContextBuilder().getParentElementOfType(CtTry.class);
+				int endOfTry = tryStatement.getPosition().getSourceEnd();
+				//offset of the bracket before catch
+				int lastBracket = getEndOfLastTryBlock(tryStatement, 0);
+				int catchStart = findNextNonWhitespace(contents, endOfTry, lastBracket + 1);
+				if (CATCH.equals(new String(contents, catchStart, CATCH.length())) == false) {
+					throw new SpoonException("Unexpected beginning of catch statement on offset: " + catchStart);
+				}
+				int bracketStart = findNextNonWhitespace(contents, endOfTry, catchStart + CATCH.length());
+				if (bracketStart < 0) {
+					throw new SpoonException("Unexpected end of file instead of \'(\' after catch statement on offset: " + catchStart);
```

Comment:
I'm afraid of this exception because there is a chance that we forget to handle one specific case and it will break the construction of the model.

Reply:
What do you suggest? To silently ignore our mistake ("we forget to handle one specific case") and to produce invalid source position?

Reply:
I don't know what is the best solution.

Reply:
I meant skipSelfChecks

Reply:
ok, I will do so